executor/executor.go

```diff
@@ -358,7 +361,8 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 			return fmt.Errorf("Updating step start %q: %v", cmd.Name, err)
 		}
 
-		code, cmdErr = doRunTeardownCommand(cmd, emitter, path, shellBin, exportFile, sourceDir)
+		code, cmdErr = doRunTeardownCommand(cmd, emitter, path, shellBin, exportFile, sourceDir, aggExitCode)
+		aggExitCode = code | aggExitCode
```

Comment:
why aggregate ? can't we just expose the latest non-zero step exit code ?

Reply:
sure